homeassistant/components/cover/legrandinone.py

```diff
@@ -0,0 +1,122 @@
+"""
+Support for legrandinone Cover devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/cover.legrandinone/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.legrandinone import (
+    DATA_ENTITY_LOOKUP, CONF_MEDIA, CONF_COMM_MODE,
+    DEVICE_DEFAULTS_SCHEMA, EVENT_KEY_COMMAND,
+    DEVICE_TYPE_AUTOMATION, DATA_DEVICE_REGISTER,
+    EVENT_KEY_ID, LegrandInOneCommand)
+from homeassistant.components.cover import (
+    CoverDevice, PLATFORM_SCHEMA)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_NAME
+
+
+DEPENDENCIES = ['legrandinone']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+CONF_DEVICE_DEFAULTS = 'device_defaults'
+CONF_DEVICES = 'devices'
+CONF_AUTOMATIC_ADD = 'automatic_add'
+CONF_FIRE_EVENT = 'fire_event'
+CONF_RECONNECT_INTERVAL = 'reconnect_interval'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_DEVICE_DEFAULTS, default=DEVICE_DEFAULTS_SCHEMA({})):
+    DEVICE_DEFAULTS_SCHEMA,
+    vol.Optional(CONF_AUTOMATIC_ADD, default=True): cv.boolean,
+    vol.Optional(CONF_DEVICES, default={}): vol.Schema({
+        cv.string: {
+            vol.Optional(CONF_NAME): cv.string,
+            vol.Optional(CONF_FIRE_EVENT, default=False): cv.boolean,
+            vol.Optional(CONF_MEDIA, default='plc'): cv.string,
+            vol.Optional(CONF_COMM_MODE, default='unicast'): cv.string,
+        },
+    }),
+})
+
+
+def devices_from_config(domain_config):
+    """Parse configuration and add IOBL cover devices."""
+    devices = []
+    for device_id, config in domain_config[CONF_DEVICES].items():
+        device_config = dict(domain_config[CONF_DEVICE_DEFAULTS], **config)
+        device = LegrandInOneCover(device_id, **device_config)
+        devices.append(device)
+
+    return devices
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the IOBL cover platform."""
+    async_add_entities(devices_from_config(config))
+
+    async def add_new_device(event):
+        """Check if device is known, otherwise add to list of known devices."""
+        device_id = event[EVENT_KEY_ID]
+
+        device_config = config[CONF_DEVICE_DEFAULTS]
+        device = LegrandInOneCover(device_id, initial_event=event,
+                                   **device_config)
+        async_add_entities([device])
+        hass.data[DATA_ENTITY_LOOKUP][
+            EVENT_KEY_COMMAND][device_id].append(device)
```

Comment:
The entity should not be stored in a shared container. Entities are only allowed to be accessed directly from their platforms. It's ok to store a reference to the entity if needed, eg the entity_id. We already use the dispatch helper so accessing the entity directly shouldn't be needed.

Reply:
Looks like it's not required. I removed it.